subway_prediction_ml/terminal_modes_adder.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from location_id_translator import LocationIdTranslator
+
+
+class TerminalModesAdder(BaseEstimator, TransformerMixin):
+    def __init__(self, locations_path, patterns_path, terminals_path):
+        self.locations_path = locations_path
+        self.patterns_path = patterns_path
+        self.terminals_path = terminals_path
+
+    def fit(self, vehicle_datapoints, y=None):
+        return self
+
+    def transform(self, vehicle_datapoints, y=None):
+        terminals = self._gtfs_id_for_terminals()
+
+        frame_with_terminals = pd.merge(
+            vehicle_datapoints,
+            right=terminals,
+            left_on="pattern_id",
+            right_on="pattern_id",
+            how="inner"
+        )
+        frame_with_terminal_modes = pd.merge(
```

Comment:
@losvedir I'll look into that

Reply:
@losvedir I don't know why this is happening yet but there's one big clue:

```
(Pdb) frame_with_terminal_modes.terminal_gtfs_id.value_counts()
70061    638
70105    535
70001    436
70036    436
Name: terminal_gtfs_id, dtype: int64
```

If these four terminal IDs look familiar, it's because they're the parent stop IDs for our four multi-platform terminals.

Reply:
I just confirmed that the same thing happens with a different and much larger dataset, so it's not some kind of fluke in your dataset.

Reply:
@losvedir I just pushed a commit that should address this